src/controllers/publish.ts

```diff
@@ -466,8 +467,15 @@ export const downloadDataset = async (req: Request, res: Response, next: NextFun
       revision = singleLangRevision(revWithMeta, req.language)!;
     }
 
+    let attachmentName: string;
+    if (revision.metadata?.title) {
+      attachmentName = `${slugify(revision.metadata?.title, { lower: true })}-${revision.revision_index > 0 ? `v${revision.revision_index}` : 'draft'}`;
+    } else {
+      attachmentName = `${dataset.id}-${revision.revision_index > 0 ? `v${revision.revision_index}` : 'draft'}`;
+    }
```

Comment:
Can we extract this to a helper function? Or move it to `getDownloadHeaders`?

Reply:
We can probably move it around... I was planning to revisit it when I redo the downloads for the upcoming consumer view tickets.